aries_cloudagent/vc/data_integrity/cryptosuites/eddsa_jcs_2022.py

```diff
@@ -0,0 +1,201 @@
+                "created"
+            ], "Expected proof.created to be a [XMLSCHEMA11-2] dateTimeStamp string."
+
+        if "expires" in proof_config:
+            # TODO assert proper [XMLSCHEMA11-2] dateTimeStamp string
+            assert proof_config[
+                "expires"
+            ], "Expected proof.expires to be a [XMLSCHEMA11-2] dateTimeStamp string."
+
+        return self._canonicalize(proof_config)
+
+    def transformation(self, unsecured_document: dict, options: dict):
+        """Transformation algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#transformation-eddsa-jcs-2022.
+        """
+        assert (
+            options["type"] == "DataIntegrityProof"
+        ), "Expected proof.type to be `DataIntegrityProof`"
+        assert (
+            options["cryptosuite"] == "eddsa-jcs-2022"
+        ), "Expected proof.cryptosuite to be `eddsa-jcs-2022`"
+
+        return self._canonicalize(unsecured_document)
+
+    def hashing(self, transformed_document, canonical_proof_config):
+        """Hashing algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#hashing-eddsa-jcs-2022.
+        """
+        return (
+            sha256(canonical_proof_config).digest()
+            + sha256(transformed_document).digest()
+        )
+
+    async def proof_serialization(self, hash_data: bytes, options: dict):
+        """Proof Serialization Algorithm.
+
+        https://www.w3.org/TR/vc-di-eddsa/#proof-serialization-eddsa-jcs-2022.
+        """
+        # If the verification method is a did:key: URI,
+        # we derive the signing key from a multikey value
+        if options["verificationMethod"].startswith("did:key:"):
+            multikey = options["verificationMethod"].split("#")[-1]
+            key_info = await self.key_manager.from_multikey(multikey)
+
+        # Otherwise we derive the signing key from a kid
+        else:
+            key_info = await self.key_manager.from_kid(options["verificationMethod"])
+
+        return await self.wallet.sign_message(
+            message=hash_data,
+            from_verkey=self.key_manager._multikey_to_verkey(key_info["multikey"]),
+        )
+
+    def _canonicalize(self, data: dict):
+        """Json canonicalization."""
+        return canonicaljson.encode_canonical_json(data)
+
+    async def _get_multikey(self, kid: str):
+        """Derive a multikey from the verification method."""
+
+        # If verification method is a did:key URI,
+        # we derive the multikey directly from the value.
+        if kid.startswith("did:key:"):
+            return kid.split("#")[-1]
+
+        # Otherwise we resolve the verification method and extract the multikey.
+        else:
+            verification_method = await DIDResolver().dereference(
+                profile=self.session.profile, did_url=kid
+            )
+            assert (
+                verification_method["type"] == "Multikey"
+            ), "Expected Multikey verificationMethod type"
```

Comment:
This seems like an inappropriate use of an assertion. Do we have control over whether the resolved verification method will always be a `Multikey`? If not, this should be a more specific exception.

Reply:
The spec defines the verification method as Multikey
https://www.w3.org/TR/vc-di-eddsa/#verification-methods

It seems like a fair assertion.

Reply:
Very interesting... That will have some fun DID Method interop implications. Even given the spec requirement, I get the feeling a more specific exception would be beneficial because there is still the opportunity for the VM type to not be a Multikey, isn't there?

An exception denoting the expectations of the spec have been violated is different than an assertion error denoting the expectations of the programmer/program have been violated. I would apply these same comments to the other instances of assertions being used in this PR.

Reply:
I decided to accept a couple more verification method and no longer restrict on multikey